### PR TITLE
Update javadoc

### LINKED ISSUE #1181: Unify Javadoc for configuration classes
With #996, [Span|Metric] Processors and Exporters can accept configuration parameters from different sources. The documentation that lists the names for the names used to do the look-up is duplicated among the different classes. This documentation should be unified in the class-level Javadoc and in the package-info.

### PR DESCRIPTION
**Pull request by thisthat**
This PR resolves #1181

### CODE REVIEW

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java [hunk 1]
```diff
@@ -30,7 +30,27 @@
 import java.util.logging.Logger;
 import javax.annotation.concurrent.ThreadSafe;
 
-/** Exports metrics using OTLP via gRPC, using OpenTelemetry's protobuf model. */
+/**
+ * Exports metrics using OTLP via gRPC, using OpenTelemetry's protobuf model.
+ *
+ * <p>Values for {@link OtlpGrpcMetricExporter} can be read from system properties, environment
```
Comment (carlosalberto) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java:36]: `Options` or `configuration values` perhaps?

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java [hunk 2]
```diff
@@ -30,7 +30,27 @@
 import java.util.logging.Logger;
 import javax.annotation.concurrent.ThreadSafe;
 
-/** Exports spans using OTLP via gRPC, using OpenTelemetry's protobuf model. */
+/**
+ * Exports spans using OTLP via gRPC, using OpenTelemetry's protobuf model.
+ *
+ * <p>Values for {@link OtlpGrpcSpanExporter} can be read from system properties, environment
```
Comment (jkwatson) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java:36]: s/Values/Configuration options/

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java [hunk 3]
```diff
@@ -36,6 +36,22 @@
  * Wraps a list of {@link MetricProducer}s and automatically reads and exports the metrics every
  * export interval.
  *
+ * <p>Values for {@link IntervalMetricReader} can be read from system properties, environment
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java:39]: s/Values/Configuration options/

File: sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java [hunk 4]
```diff
@@ -36,6 +36,22 @@
...
+ * variables, or {@link java.util.Properties} objects.
+ *
+ * <p>For System Properties and {@link java.util.Properties} objects, {@link IntervalMetricReader}
+ * will look for the following names:
+ *
+ * <ul>
+ *   <li>{@code otel.imr.export.interval}: sets the export interval between pushes to the exporter.
+ * </ul>
+ *
+ * <p>For Environment Variable, {@link IntervalMetricReader} will look for the following names:
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/metrics/export/IntervalMetricReader.java:49]:
```suggestion
 * <p>For environment variables, {@link IntervalMetricReader} will look for the following names:
```

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java [hunk 5]
```diff
@@ -30,7 +30,27 @@
...
+ *
+ * <p>For System Properties and {@link java.util.Properties} objects, {@link OtlpGrpcSpanExporter}
+ * will look for the following names:
+ *
+ * <ul>
+ *   <li>{@code otel.otlp.span.timeout}: to set the max waiting time for the collector to process
+ *       each span batch.
+ * </ul>
+ *
+ * <p>For Environment Variable, {@link OtlpGrpcSpanExporter} will look for the following names:
```
Comment (jkwatson) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java:47]:
```suggestion
 * <p>For environment variables, {@link OtlpGrpcSpanExporter} will look for the following names:
```

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java [hunk 6]
```diff
@@ -30,7 +30,27 @@
...
+ *
+ * <p>For System Properties and {@link java.util.Properties} objects, {@link OtlpGrpcMetricExporter}
+ * will look for the following names:
+ *
+ * <ul>
+ *   <li>{@code otel.otlp.metric.timeout}: to set the max waiting time for the collector to process
+ *       each metric batch.
+ * </ul>
+ *
+ * <p>For Environment Variable, {@link OtlpGrpcMetricExporter} will look for the following names:
```
Comment (jkwatson) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java:47]:
```suggestion
 * <p>For environment variables, {@link OtlpGrpcMetricExporter} will look for the following names:
```

File: sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java [hunk 7]
```diff
@@ -56,6 +56,30 @@
  *
  * <p>This batch {@link SpanProcessor} can cause high contention in a very high traffic service.
  * TODO: Add a link to the SpanProcessor that uses Disruptor as alternative with low contention.
+ *
+ * <p>Values for {@link BatchSpansProcessor} can be read from system properties, environment
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java:60]: s/Values/Configuration options/

File: sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java [hunk 8]
```diff
@@ -56,6 +56,30 @@
...
+ *
+ * <ul>
+ *   <li>{@code otel.bsp.schedule.delay}: sets the delay interval between two consecutive exports.
+ *   <li>{@code otel.bsp.max.queue}: sets the maximum queue size.
+ *   <li>{@code otel.bsp.max.export.batch}: sets the maximum batch size.
+ *   <li>{@code otel.bsp.export.timeout}: sets the maximum allowed time to export data.
+ *   <li>{@code otel.bsp.export.sampled}: sets whether only sampled spans should be exported.
+ * </ul>
+ *
+ * <p>For Environment Variable, {@link BatchSpansProcessor} will look for the following names:
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java:74]:
```suggestion
 * <p>For environment variables, {@link BatchSpansProcessor} will look for the following names:
```

File: sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSpansProcessor.java [hunk 9]
```diff
@@ -32,6 +32,22 @@
 /**
  * An implementation of the {@link SpanProcessor} that converts the {@link ReadableSpan} to {@link
  * SpanData} and passes it to the configured exporter.
+ *
+ * <p>Values for {@link SimpleSpansProcessor} can be read from system properties, environment
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSpansProcessor.java:36]: s/Values/Configuration options/

File: sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSpansProcessor.java [hunk 10]
```diff
@@ -32,6 +32,22 @@
...
+ * variables, or {@link java.util.Properties} objects.
+ *
+ * <p>For System Properties and {@link java.util.Properties} objects, {@link SimpleSpansProcessor}
+ * will look for the following names:
+ *
+ * <ul>
+ *   <li>{@code otel.ssp.export.sampled}: sets whether only sampled spans should be exported.
+ * </ul>
+ *
+ * <p>For Environment Variable, {@link SimpleSpansProcessor} will look for the following names:
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSpansProcessor.java:46]:
```suggestion
 * <p>For environment variables, {@link SimpleSpansProcessor} will look for the following names:
```

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java [hunk 11]
```diff
@@ -30,7 +30,27 @@
...
-/** Exports metrics using OTLP via gRPC, using OpenTelemetry's protobuf model. */
+/**
+ * Exports metrics using OTLP via gRPC, using OpenTelemetry's protobuf model.
+ *
+ * <p>Configuration options for {@link OtlpGrpcMetricExporter} can be read from system properties,
+ * environment variables, or {@link java.util.Properties} objects.
+ *
+ * <p>For System Properties and {@link java.util.Properties} objects, {@link OtlpGrpcMetricExporter}
+ * will look for the following names:
+ *
```
Comment (bogdandrutu) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java:41]: s/For System Properties/For system properties

And everywhere

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java [hunk 12]
```diff
@@ -30,7 +30,27 @@
...
+ * <ul>
+ *   <li>{@code otel.otlp.metric.timeout}: to set the max waiting time for the collector to process
+ *       each metric batch.
```
Comment (bogdandrutu) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java:44]: "to set the max waiting time allowed to send each metric batch"?

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java [hunk 13]
```diff
@@ -30,7 +30,27 @@
...
+ * Exports spans using OTLP via gRPC, using OpenTelemetry's protobuf model.
+ *
+ * <p>Configuration options for {@link OtlpGrpcSpanExporter} can be read from system properties,
+ * environment variables, or {@link java.util.Properties} objects.
+ *
+ * <p>For System Properties and {@link java.util.Properties} objects, {@link OtlpGrpcSpanExporter}
+ * will look for the following names:
+ *
+ * <ul>
+ *   <li>{@code otel.otlp.span.timeout}: to set the max waiting time for the collector to process
```
Comment (bogdandrutu) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java:43]: "to set the max waiting time allowed to send each span batch"?

**Pull request merged**